Fix: (Accessibility) replace 'Show More' glyph label with 'Menu' in mobile Metabar

ZallaxDev opened the pull request:
The "Show More" glyph button in the mobile view of the Metabar has been replaced
with a descriptive text label "Menu".
This change improves accessibility and clarity for users, following the recommendations in ticket #[41471](https://mantis.ilias.de/view.php?id=41471).

The code for the tests in `NotificationItemTest.html` and `MetaBarTest.php` has also been modified by adapting the language variable.

ZallaxDev committed:
Fix: replace 'Show More' glyph label with 'Menu' in mobile Metabar

The "Show More" glyph button in the mobile view of the Metabar has been replaced
with a descriptive text label "Menu".
This change improves accessibility and clarity for users, following the recommendations in ticket #41471.

ZallaxDev committed:
Fix: Recovered original show_more

In the previous commit, you replaced the entry “show_more” with “menu_label”.
In this commit, we restore the entry “show_more” to avoid problems in case this label is used in other parts of ILIAS.

ZallaxDev committed:
Tests: update MetaBar and ItemNotification expectations for “Menu” label

Fix failing unit tests after ticket #41471, which replaced the mobile Metabar
toggle label from “show_more” to the new “Menu” label (menu_label).

Updated expected HTML strings in:
- components/ILIAS/UI/tests/Component/MainControls/MetaBarTest.php
  (MetaBarTest::testRendering)
- components/ILIAS/UI/tests/Component/Item/ItemNotificationClientHtmlTest.php
  (ItemNotificationClientHtmlTest::testRenderClientHtml)

The tests now assert the new label value (menu_label / “Menu”) instead of the
previous “show_more”. ARIA attributes and structure remain unchanged.

ZallaxDev committed:
Lang: reorder language variables for consistency

Reordered language variables to maintain consistent alphabetical structure
and improve readability across translation files.

ZallaxDev committed:
feat: add German translation for "Menu" label

ZallaxDev commented:
Since this is the first time we are including a new language variable, I would like to mention @Chris-Squirrel to help us with the procedure.

We have currently integrated these variables into German and English.

How should we proceed with the inclusion in the other languages?

Thank you very much in advance!

Chris-Squirrel commented:
@ZallaxDev 

Hi - most of the new variables are only present in English and German. I have conferred with Fabian Wolf and he will update all the  languages tomorrow after(?) 'coding complete'. Then the variables for Spanish, French, etc. should be available with the English entries as default. These can then be updated according to normal procedures.

klees reviewed (changes requested):
Hi @ZallaxDev,

thanks for looking into this. This looks plausible.

Please implement the following changes:
- [x] lang-var: Please move it to the namespace of the UI framework ("ui") and name it specifically like its usage location ("metabar_slate_label").

Also, we want to point out that this only can have a very broad name (like "Menu") because during discussion of the page layout revision we lost conceptual clarity regarding the semantics of the Meta Bar and instead ended up putting all stuff there that anyone wanted to have in the Meta Bar badly enough. The initial concept from @catkrahl and @yseiler planned to have all items in the Meta Bar where the system pushes info to the user. This would have allowed for a more pointed label like "Notification Menu" or something like that. There might be other options for a more defined concept of the Main Bar, but "Menu" seems to be the best description for the current state of the art.

None of this is your fault, ofc, but we feel this is a nice opportunity to highlight how weak concepts directly weaken how well UIs can be understood by users.

Best regards!
@yvseiler, @thibsy and @klees

ZallaxDev commented:
Hello @Chris-Squirrel and @klees,

Thank you very much for your helpful comments.
We’ll implement the changes proposed by @klees and ensure that the English updates mentioned by @Chris-Squirrel are applied consistently across the other languages.

Once again, thank you for your valuable feedback. I’ll let you know as soon as the revisions are ready for review.

Best regards,

klees commented:
Hi @ZallaxDev,

please assign and ping me once you are done.

Thank you!

ZallaxDev committed:
update: Moved the new language variable to the namespace of the UI framework and renamed it. Added to all languages.

The new language variable has been moved to the namespace of the UI framework and renamed.
Added to all languages using English and adding the date added.

ZallaxDev commented:
Hello @klees, @Chris-Squirrel, @thibsy and @yvseiler:

We have reorganized the new language variable to the namespace of the UI framework and renamed it as indicated.

In addition, we have included the new language variable in all languages, using the English variable and adding the date it was added (following other variables that have not yet been processed).

We look forward to your feedback.

Thank you all very much for your time!

Best regards,
Daniel.

Chris-Squirrel commented:
Looks good to me!

Annett7811 reviewed (approved):
From an accessibility perspective, this looks good. Thank you very much for your work.

Best regards,
Annett

oliversamoila commented:
Hello @ZallaxDev.
Thank you very much for your contribution and your PR. Please also provide PRs for releases 9, 11 and the trunk.
Please also resolve the conflicts.
Many thanks for your support. 
Best regards,
@oliversamoila

ZallaxDev commented:
Good afternoon everyone:
First of all, I would like to thank @Annett7811 and @oliversamoila for their comments and feedback 😊

> Hello @ZallaxDev. Thank you very much for your contribution and your PR. Please also provide PRs for releases 9, 11 and the trunk. Please also resolve the conflicts. Many thanks for your support. Best regards, @oliversamoila

I take note of your instructions. We will review the code conflict and make the changes in the indicated branches.

We will report back via this PR to notify you of the conflict resolution.

Thank you for your time.

Best regards,
Daniel.

juanma1331 committed:
Fix(Accessibility): Replace 'Show More' glyph label with 'Menu' in mobile
     Metabar. Ticket 41471

juanma1331 committed:
Update: Italian language file with latest translation updates

juanma1331 committed:
Add: Italian translation for metabar_slate_label variable.
Added ui#:#metabar_slate_label#:#Menu###07 11 2025 new variable to Italian language file.
This variable provides the 'Menu' label for the MetaBar mobile view accessibility fix.

ZallaxDev committed:
Merge branch 'ilias10_a11y_M41471' of https://github.com/surlabs/ILIAS into release_10

ZallaxDev committed:
Merge branch 'ILIAS-eLearning:release_10' into ilias10_a11y_M41471

ZallaxDev commented:
Hello,

We’ve applied the requested changes and resolved the conflicts with the Italian translation.
@klees I’m mentioning and assigning you as you indicated.

We’re now waiting for your review, and if any additional changes are needed, we’re more than happy to help.

Thanks everyone 😊

Best regards,
Daniel

oliversamoila commented:
Hello @ZallayDev, 
I will take over here for the next step, as responsibilities have changed since the last review. I will then hand over to Thibeau Fuhrer (thibsy) for a code review.

My comments will follow here accordingly.
Best regards,
Oliver

oliversamoila reviewed (approved):
Hello together, 
from an UX and UI perspective, the result looks good. Thank you very much for your contribution, @ZallaxDev.
@Annett7811 has already approved it. Thank you as well.
I'm forwarding this to @thibsy for code review. He may contact you with change requests or incorporate the changes directly (That includes cherry picking too.)

Kind regards,
@oliversamoila (as UI coordinator)

abrahammordev commented:
Hi @thibsy,

After resolving several conflicts, I have updated the PR with the latest language files.
Please let me know if you need anything else from our side.

Best,
Abraham
